### PR TITLE
Merge dev into master (release 0.26.0)

### PR DESCRIPTION
**Pull request by Pronyo-Chan**

### CODE REVIEW

File: models/options/OptionSet.py [hunk 1]
```diff
@@ -383,14 +387,19 @@ def update_options(self, options_dict=None):
         # Starting setup
         if "StartingMap" in options_dict:
             self.starting_map = options_dict.get("StartingMap")
-        if "StartingLevel" in options_dict:
-            self.starting_level = options_dict.get("StartingLevel")
+        #if "StartingLevel" in options_dict:
```
Comment (Pronyo-Chan) [inline models/options/OptionSet.py:390]: Should we just delete that?

File: models/options/OptionSet.py [hunk 2]
```diff
@@ -1671,3 +1685,373 @@ def get_startitem_list(self) -> list:
...
+                "Kooper": self.start_with_kooper,
+                "Bombette": self.start_with_bombette,
+                "Parakarry": self.start_with_parakarry,
+                "Bow": self.start_with_bow,
+                "Watt": self.start_with_watt,
+                "Sushie": self.start_with_sushie,
+                "Lakilester": self.start_with_lakilester,
+            }
+
+        #if WriteSpoilerLog and RevealLogInHours != 0:
```
Comment (Pronyo-Chan) [inline models/options/OptionSet.py:1873]: Can be cleaned up

File: models/options/OptionSet.py [hunk 3]
```diff
@@ -1671,3 +1685,373 @@ def get_startitem_list(self) -> list:
...
+        web_settings["FastFloodRoomBombetteUltraBoots"] = self.glitch_settings.fast_flood_room_bombette_ultra_boots
+
+        # Glitches: Global
+        web_settings["BreakStoneBlocksWithUltraBoots"] = self.glitch_settings.break_stone_blocks_with_ultra_boots
+        web_settings["BreakYellowBlocksWithSuperBoots"] = self.glitch_settings.break_yellow_blocks_with_super_boots
+        web_settings["KnowsHiddenBlocks"] = self.glitch_settings.knows_hidden_blocks
+        web_settings["KnowsPuzzleSolutions"] = self.glitch_settings.knows_puzzle_solutions
+        web_settings["ReachHighBlocksWithSuperBoots"] = self.glitch_settings.reach_high_blocks_with_super_boots
+
+        # Other/Hidden Options
```
Comment (Pronyo-Chan) [inline models/options/OptionSet.py:2051]: Same, can be deleted

File: presets/preset_ExtremeShuffle.yaml [hunk 4]
```diff
@@ -169,7 +168,8 @@ RandomPartnerFP  : 3
 RandomStarpowerSP: 3
 
 # Misc Gameplay Randomization
-ShuffleBlocks: true
+ShuffleBlocks   : true
+RandomizePuzzles: false
```
Comment (Pronyo-Chan) [inline presets/preset_ExtremeShuffle.yaml:172]: Should we randomize puzzles in the extreme preset?

File: rando_modules/random_puzzles_minigames.py [hunk 5]
```diff
@@ -0,0 +1,620 @@
...
+            puzzle_minigame_list.append((puzzle.get_key(), positions_encoded))
+
+        # Flower Fields Three Tree: Correct hit sequence
+        elif puzzle.name == "FlowerFieldsThreeTrees":
+            if not random_puzzles:
+                sequence_encoded = puzzle.default_value
+                spoilerlog_additions["FlowerFieldsThreeTrees"] = "Middle, Right, Left"
+            else:
+                def _map_tree(tree_id: int) -> str:
+                    if tree_id == 1:
```
Comment (Pronyo-Chan) [inline rando_modules/random_puzzles_minigames.py:316]: Shouldn't it be if tree_id ==0?

File: models/options/OptionSet.py (hunk 1 above)
Comment (Pronyo-Chan) [reply models/options/OptionSet.py]: done

File: models/options/OptionSet.py (hunk 2 above)
Comment (Pronyo-Chan) [reply models/options/OptionSet.py]: done

File: models/options/OptionSet.py (hunk 3 above)
Comment (Pronyo-Chan) [reply models/options/OptionSet.py]: done

File: presets/preset_ExtremeShuffle.yaml (hunk 4 above)
Comment (icebound777) [reply presets/preset_ExtremeShuffle.yaml:172]: Sure, sounds like that would be in the spirit of extreme

File: rando_modules/random_puzzles_minigames.py (hunk 5 above)
Comment (icebound777) [reply rando_modules/random_puzzles_minigames.py:316]: Doesn't really matter, as long as the mod can handle whatever we feed it. For this one I opted for just numbering them naturally, like "tree to hit number 1, number 2, number 3"

**Pull request merged**